examples/email-recovery/src/pages/index.tsx

```diff
@@ -0,0 +1,239 @@
+import Image from "next/image";
+import styles from "./index.module.css";
+import { getWebAuthnAttestation, TurnkeyClient } from "@turnkey/http";
+import { IframeStamper } from "@turnkey/iframe-stamper";
+import { useForm } from "react-hook-form";
+import axios from "axios";
+import * as React from "react";
+import { useState } from "react";
+import { Recovery } from "@/components/Recovery";
+
+/**
+ * Type definition for the server response coming back from `/api/initRecovery`
+ */
+type InitRecoveryResponse = {
+  userId: string;
+  organizationId: string;
+};
+
+/**
+ * Type definitions for the form data (client-side forms)
+ */
+type RecoverUserFormData = {
+  recoveryBundle: string;
+  authenticatorName: string;
+};
+type InitRecoveryFormData = {
+  targetPublicKey: string;
+  email: string;
+};
+
+// All algorithms can be found here: https://www.iana.org/assignments/cose/cose.xhtml#algorithms
+// We only support ES256 and RS256, which are listed here
+const es256 = -7;
+const rs256 = -257;
+
+// This constant designates the type of credential we want to create.
+// The enum only supports one value, "public-key"
+// https://www.w3.org/TR/webauthn-2/#enumdef-publickeycredentialtype
+const publicKey = "public-key";
+
+const generateRandomBuffer = (): ArrayBuffer => {
+  const arr = new Uint8Array(32);
+  crypto.getRandomValues(arr);
+  return arr.buffer;
+};
+
+const base64UrlEncode = (challenge: ArrayBuffer): string => {
+  return Buffer.from(challenge)
+    .toString("base64")
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_")
+    .replace(/=/g, "");
+};
+
+export default function RecoveryPage() {
+  const [initRecoveryResponse, setInitRecoveryResponse] =
+    useState<InitRecoveryResponse | null>(null);
+  const [iframeStamper, setIframeStamper] = useState<IframeStamper | null>(
+    null
+  );
+  const {
+    register: initRecoveryFormRegister,
+    handleSubmit: initRecoveryFormSubmit,
+  } = useForm<InitRecoveryFormData>();
+  const {
+    register: recoverUserFormRegister,
+    handleSubmit: recoverUserFormSubmit,
+  } = useForm<RecoverUserFormData>();
+
+  const initRecovery = async (data: InitRecoveryFormData) => {
+    const response = await axios.post("/api/initRecovery", {
+      email: data.email,
+      targetPublicKey: data.targetPublicKey,
+    });
+    setInitRecoveryResponse(response.data);
+  };
+
+  const recoverUser = async (data: RecoverUserFormData) => {
+    if (iframeStamper === null) {
```

Comment:
if we want to cover the `undefined` case, we can do a `falsy` check here (and in the line to follow for `initRecoveryResponse`)

Reply:
Sounds good to me!